test/acl.test.js

```diff
@@ -75,23 +77,22 @@ describe('security scopes', function() {
             model: 'testModel', property: 'name',
             accessType: ACL.WRITE, permission: ACL.DENY,
           }, function(err, resource) {
-          // console.log(resource);
-            Scope.checkPermission('testModelScope', 'testModel', ACL.ALL, ACL.ALL,
-              function(err, perm) {
-                assert(perm.permission === ACL.DENY); // because name.WRITE == DENY
-              });
-            Scope.checkPermission('testModelScope', 'testModel', 'name', ACL.ALL,
-              function(err, perm) {
-                assert(perm.permission === ACL.DENY); // because name.WRITE == DENY
-              });
-            Scope.checkPermission('testModelScope', 'testModel', 'name', ACL.READ,
-              function(err, perm) {
-                assert(perm.permission === ACL.ALLOW);
-              });
-            Scope.checkPermission('testModelScope', 'testModel', 'name', ACL.WRITE,
-              function(err, perm) {
-                assert(perm.permission === ACL.DENY);
-              });
+            async.parallel([
+              clbk => Scope.checkPermission('testModelScope', 'testModel', ACL.ALL, ACL.ALL, clbk),
+              clbk => Scope.checkPermission('testModelScope', 'testModel', 'name', ACL.ALL, clbk),
+              clbk => Scope.checkPermission('testModelScope', 'testModel', 'name', ACL.READ, clbk),
+              clbk => Scope.checkPermission('testModelScope', 'testModel', 'name', ACL.WRITE, clbk),
+            ], (err, perms) => {
+              assert(!err);
+              assert(perms.length === 4);
+              if (perms.length === 4) {
+                assert(perms[0].permission === ACL.DENY); // because name.WRITE == DENY
+                assert(perms[1].permission === ACL.DENY); // because name.WRITE == DENY
+                assert(perms[2].permission === ACL.ALLOW);
+                assert(perms[3].permission === ACL.DENY);
```

Comment:
Please use `assert.equal` and add a custom message explaining which scenario is being asserted on, so that the test failure tells what exactly went wrong.

Alternatively, and I think this is a better solution, assert on the entire content of the array, for example:

```js
const actualPermissions = perms.map(p => p.permission);
assert.deepEqual(actualPermissions, [
  ACL.DENY, // because name.WRITE == DENY
  ACL.DENY, // // because name.WRITE == DENY
  ACL.ALLOW,
  ACL.DENY
]);
```

That way you don't need to assert on the length of the array. Instead of a not very helpful error "expected 3 to be 4", `deepEqual` will show the full array with permission values.